### PR TITLE
Setting to show new post after reply

### PR DESCRIPTION
**Pull request by boomzillawtf**
User configurable setting: after posting a reply show the new post in the browser or stay at the current browsing position.

Related:
https://github.com/NodeBB/nodebb-theme-vanilla/pull/69
https://github.com/NodeBB/nodebb-theme-persona/pull/252

### CODE REVIEW

File: public/src/client/topic/posts.js [hunk 1]
```diff
@@ -33,10 +33,12 @@ define('forum/topic/posts', [
 		ajaxify.data.postcount ++;
 		postTools.updatePostCount(ajaxify.data.postcount);
 
-		if (config.usePagination) {
-			onNewPostPagination(data);
-		} else {
-			onNewPostInfiniteScroll(data);
+		if (config.scrollToMyPost) {
```
Comment (barisusakli) [inline public/src/client/topic/posts.js]: Can you change this to `ajaxify.data.scrollToMyPost` and add `scrollToMyPost` here https://github.com/NodeBB/NodeBB/blob/master/src/controllers/topics.js#L261 so it is passed to the client.

**Comment by barisusakli** [comment]
Looks good so far, can you also put the links to the persona/vanilla changes here ?

File: public/src/client/topic/posts.js (hunk 1 above)
Comment (boomzillawtf) [inline public/src/client/topic/posts.js]: Changing to ajaxify.data is easy, though the way it is now seems to work. I don't understand why this is better (but I'll be honest, I've barely scratched the surface of figuring out how everything fits together).

Also, I've been looking and I have no clue how to make the change in controllers/topics.js or why it's necessary. Also, the change seems to get passed to the client already.

Comment (barisusakli) [inline public/src/client/topic/posts.js]: Hmm I don't see anywhere that you pass scrollToMyPost the client. We pass them to the client here https://github.com/NodeBB/NodeBB/blob/master/src/controllers/api.js#L73 if we need the setting to be available in `config.*` but in this case we only need it in the topic view, that's why I suggested adding it to the topics controller.

I can make the change after I merge.

Comment (barisusakli) [inline public/src/client/topic/posts.js]: https://github.com/NodeBB/NodeBB/commit/9e44bd3d219849267e4f2b6514fc388cd0f6c286

**Comment by pichalite** [comment]
so this is going to surprise every user on existing installations after upgrade. the user setting should have been the other way around to not surprise existing users and look like something broke.

**Pull request merged**